### PR TITLE
Add optional debug_message for all Errors

### PR DESCRIPTION
**Pull request by riiwo**
These values will be used in FE to determine correct error to display.

### CODE REVIEW

File: lib/salestation/app/input_verification.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def verify_input(schema)
             request.replace_input(result.output)
           else
             Deterministic::Result::Failure(
-              Errors::InvalidInput.new(errors: result.errors, hints: result.hints)
+              Errors::InvalidInput.new(errors: result.errors, hints: result.hints, debug_message: result.debug_message)
```
Comment (indrekj) [inline lib/salestation/app/input_verification.rb]: can you add debug_message only if it's present?

Comment (indrekj) [reply lib/salestation/app/input_verification.rb]: nvm

**Comment by urmastalimaa** [comment]
> These values will be used in FE to determine correct error to display.

If this is the goal, then I think it's much better to _finally_ start with adding error codes instead of debug_message. For this we need some kind of structure for the error codes, which will need discussion with @indrekj and others, *but* I feel that if we don't start with that, we never will.

File: spec/salestation/app/errors/dependency_currently_unavailable_spec.rb [hunk 2]
```diff
@@ -8,4 +8,16 @@
   it 'has message' do
     expect(create_dependency_currently_unavailable.message).to eq(message)
   end
+
+  context 'when debug_message is not required' do
```
Comment (urmastalimaa) [inline spec/salestation/app/errors/dependency_currently_unavailable_spec.rb]: I don't understand the description of the `context`. I don't see anything in the context `subject` nor `let`s that communicates `debug_message` being not required to me.

**Comment by riiwo** [comment]
> > These values will be used in FE to determine correct error to display.
> 
> If this is the goal, then I think it's much better to _finally_ start with adding error codes instead of debug_message. For this we need some kind of structure for the error codes, which will need discussion with @indrekj and others, **but** I feel that if we don't start with that, we never will.

Well currently we have debug_messages that are used in FE for most of the errors. But also some, that are used in creating observation for example, do not support that, and it ends up defaulting to network error in FE. I wanted to implement that for now to actually get meaningful responses in FE and make it possible to debug and triage issues. I agree that there are better solutions but can we do it later since, the change you are proposing will take much longer to implement.

**Comment by riiwo** [comment]
!squash

**Pull request merged**